app/cdap/components/NamespaceAdmin/SourceControlManagement/SourceControlManagementForm.tsx

```diff
@@ -0,0 +1,333 @@
+              'widget-attributes': {
+                default: scmAuthType[0].id,
+                layout: 'inline',
+                options: scmAuthType,
+              },
+            }}
+            onChange={(val) => {
+              handleValueChange(val, 'type');
+            }}
+          />
+          {formState.config?.auth?.type === scmAuthType[0].id && (
+            <>
+              <PropertyRow
+                value={formState.config?.auth?.tokenName}
+                property={{
+                  name: 'tokenName',
+                  description: T.translate(`${PREFIX}.auth.pat.tokenNameHelperText`).toString(),
+                  label: T.translate(`${PREFIX}.auth.pat.tokenName`).toString(),
+                  required: true,
+                }}
+                onChange={(val) => {
+                  handleValueChange(val, 'tokenName');
+                }}
+                errors={
+                  !formState.config?.auth?.tokenName && formState.error
+                    ? [{ msg: T.translate('commons.requiredFieldMissingMsg').toString() }]
+                    : []
+                }
+              />
+              <PropertyRow
+                value={formState.config?.auth?.token}
+                property={{
+                  name: 'token',
+                  description: T.translate(`${PREFIX}.auth.pat.tokenHelperText`).toString(),
+                  label: T.translate(`${PREFIX}.auth.pat.token`).toString(),
+                  required: true,
+                  'widget-type': 'password',
+                }}
+                onChange={(val) => {
+                  handleValueChange(val, 'token');
+                }}
+                errors={
+                  !formState.config?.auth?.token && formState.error
+                    ? [{ msg: T.translate('commons.requiredFieldMissingMsg').toString() }]
+                    : []
+                }
+              />
+              <PropertyRow
+                value={formState.config?.auth?.username}
+                property={{
+                  name: 'username',
+                  description: T.translate(`${PREFIX}.auth.pat.usernameHelperText`).toString(),
+                  label: T.translate(`${PREFIX}.auth.pat.username`).toString(),
+                }}
+                onChange={(val) => {
+                  handleValueChange(val, 'username');
+                }}
+              />
+            </>
+          )}
+        </StyledGroup>
+      </StyledForm>
+      <StyledButtonGroup>
+        <PrimaryOutlinedLoadingButton
+          onClick={validateConnection}
+          loading={formState.loading}
+          disabled={formState.loading}
+        >
+          {T.translate(`${PREFIX}.validate.button`)}
+        </PrimaryOutlinedLoadingButton>
+        <PrimaryContainedButton onClick={handleSubmit}>
+          {T.translate(`${PREFIX}.linkConfirm`)}
+        </PrimaryContainedButton>
+      </StyledButtonGroup>
+      {!!formState.error && (
```

Comment:
Don't use `!!` to convert to boolean https://google.github.io/styleguide/tsguide.html#type-coercion-implicit